### ISSUE TITLE
Type definitions

### ISSUE THREAD

**Issue by Cnidarias** [Kinto/kinto#1789]
Hi,

so since I started working on the project very recently I found it a little difficult to follow the flow of the code at times and I think this could largely be alleviated if we included type definitions (at least for the function definitions), that way we should still remain compatible with python3.5 but it makes it a little easier.

We could of course at the same time switch over to f-Strings as mentioned in #1659 and lose support for python 3.5 but also include type definitions for non function parameters

What are your thoughts?

**Comment by leplatrem** [comment]
Yes, I would be in favor of adding type definitions! Don't hesitate to start, even if we don't do everything we keep track here of what's remaining to cover :)

**Comment by Cnidarias** [comment]
Okay great!

I will work on it shortly, the aim is to stay compatible with 3.5 for now right?

**Comment by leplatrem** [comment]
Fixed in #3668

**Comment by leplatrem** [comment]
Sorry, #3668  didn't add type annotations, just type checking
